src/funnels/Swapper.sol

```diff
@@ -73,19 +74,31 @@ contract Swapper {
 
     function setLimits(address src, address dst, uint64 hop, uint128 cap) external auth {
         limits[src][dst] = PairLimit({
-            hop:  hop,
-            zzz:  limits[src][dst].zzz,
-            cap: cap
+            hop: hop,
+            cap: cap,
+            zzz: limits[src][dst].zzz,
+            amt: limits[src][dst].amt
         });
         emit SetLimits(src, dst, hop, cap);
     }
 
     function swap(address src, address dst, uint256 amt, uint256 minOut, address callee, bytes calldata data) external auth returns (uint256 out) {
         PairLimit memory limit = limits[src][dst];
-        require(block.timestamp >= limit.zzz + limit.hop, "Swapper/too-soon");
-        limits[src][dst].zzz = uint64(block.timestamp);
 
-        require(amt <= limit.cap, "Swapper/exceeds-max-amt");
+        unchecked {
+            if (block.timestamp - limit.zzz >= limit.hop) {
+                // Reset batch
+                limit.zzz = uint64(block.timestamp);
+                limit.amt = limit.cap;
+            }
+        }
+
+        require(amt <= limit.amt, "Swapper/exceeds-max-amt");
+
+        unchecked {
+            limits[src][dst].zzz = limit.zzz;
```

Comment:
Don't think we need to update zzz on each operation.

Reply:
The idea to put it there is to make sure both will be saved in the same SSTORE (as they share the same slot) so it is just easier to do the assignment always.

Reply:
Are you sure they share the same slot? I would have thought that the first (uint64, uint128, uint64) are packed into one uint256 slot and then the last uint128 is packed into a second uint256 slot.

Reply:
Yes you are right, I need to reorder that.

Reply:
Done.

Reply:
It might be nicer to assign to zzz and amt in the same order in both unchecked blocks.
Also - did you check to see if it is is indeed done as one write operation here?

Reply:
Order changed. Anyway the slots (and setting of variables) order is one of the main things I want to review for both Swapper and Depositor and their automation contracts.
Regarding the one write operation, I didn't check it for this code but I had done in the past. I'm 99% sure with at least the optimizer it should just write once but I'll double check with this code.